### PR TITLE
Bug(rux-checkbox): Indeterminate functionality - Astro 1736

### PR DESCRIPTION
**Pull request by Full-lifey**
## Brief Description

Indeterminate functionality was not fully fleshed out within component. This PR addresses the indeterminate state

## JIRA Link

[Astro 1736](https://rocketcom.atlassian.net/browse/ASTRO-1736?atlOrigin=eyJpIjoiMDhlODM5MmE0YjNlNGU1NmIzMDhjNGFiNzM3YjllNzAiLCJwIjoiaiJ9)

## Motivation and Context

Indeterminate behavior was not the standard behavior. Previously indeterminate prop had to be true and checked had to be true. That also meant that when using forms the checkbox would submit a value as checked was true. Now if indeterminate state is true checked reverts to false as to not submit anything. 

When clicking an indeterminate checkbox it will set the indeterminate state to false and the checked state to true.

## Types of changes

-   [x] Bug fix
-   [ ] New feature
-   [ ] Breaking change

## Checklist

-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by nortonprojects — APPROVED** [review]

File: src/components/rux-checkbox/rux-checkbox.tsx [hunk 1]
```diff
@@ -39,7 +47,12 @@ export class RuxCheckbox {
     /**
      * Toggles indeterminate state of a checkbox
      */
-    @Prop({ reflect: true }) indeterminate: boolean = false
+    @Prop({ reflect: true, mutable: true }) indeterminate: boolean = false
+
+    @Watch('indeterminate')
+    resetChecked() {
+        this.checked = false
```
Comment (markacianfrani) [inline src/components/rux-checkbox/rux-checkbox.tsx:54]: This doesn't fire a change event so the parent can't update its state.

**Review by markacianfrani — CHANGES_REQUESTED** [review]
We should still be able to set `indeterminate` imperatively and hear a change event. For example:

```
    <body>
      <rux-checkbox checked></rux-checkbox>
      <input type="checkbox" checked/>
      <button>make indeterminate</button>
      <script>
        const rux = document.querySelector('rux-checkbox')
        rux.addEventListener('rux-change', (e) => {
          console.log('heard change', e);
        })
        const int = document.querySelector('input')
        int.addEventListener('change', (e) => {
          console.log('heard native change');
        })

        document.querySelector('button').addEventListener('click', () => {
          rux.indeterminate = true
          int.indeterminate = true
        })
      </script>
    </body>
```

**Review by markacianfrani — APPROVED** [review]

**Comment by markacianfrani** [comment]
> We should still be able to set `indeterminate` imperatively and hear a change event. For example:
> 
> ```
>     <body>
>       <rux-checkbox checked></rux-checkbox>
>       <input type="checkbox" checked/>
>       <button>make indeterminate</button>
>       <script>
>         const rux = document.querySelector('rux-checkbox')
>         rux.addEventListener('rux-change', (e) => {
>           console.log('heard change', e);
>         })
>         const int = document.querySelector('input')
>         int.addEventListener('change', (e) => {
>           console.log('heard native change');
>         })
> 
>         document.querySelector('button').addEventListener('click', () => {
>           rux.indeterminate = true
>           int.indeterminate = true
>         })
>       </script>
>     </body>
> ```



> We should still be able to set `indeterminate` imperatively and hear a change event. For example:
> 
> ```
>     <body>
>       <rux-checkbox checked></rux-checkbox>
>       <input type="checkbox" checked/>
>       <button>make indeterminate</button>
>       <script>
>         const rux = document.querySelector('rux-checkbox')
>         rux.addEventListener('rux-change', (e) => {
>           console.log('heard change', e);
>         })
>         const int = document.querySelector('input')
>         int.addEventListener('change', (e) => {
>           console.log('heard native change');
>         })
> 
>         document.querySelector('button').addEventListener('click', () => {
>           rux.indeterminate = true
>           int.indeterminate = true
>         })
>       </script>
>     </body>
> ```

I guess this should be a separate ticket?

**Comment by Full-lifey** [comment]
> In the Storybook demo, once I change the checked state, I can't find any control that changes the demo back to indeterminate. The indeterminate Control does not seem to do anything to the demo.

This will be looked at from the imperative implementation ticket.

**Review by amalyah — APPROVED** [review]

**Pull request merged**